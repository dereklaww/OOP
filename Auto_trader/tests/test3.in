traders
trades
add trader1 100
add trader2 100
import trader1 ABC 100
import trader2 DEF 100
sell trader1 ABC 30 3
sell trader1 ABC 0 0
amount trader1 ABC
buy trader2 ABC 20 3
buy trader2 DEF 30 3
cancel buy 0000
cancel buy 0002
cancel sell 0000
cancel sell 0000
traders
trades
trades trader trader1
trades trader trader2
trades product ABC
trades product DEF
book sell
book buy
exit