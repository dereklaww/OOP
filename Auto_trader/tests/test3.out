No traders in the market.
No trades have been completed.
Success.
Success.
Trader now has 100.00 units of ABC.
Trader now has 100.00 units of DEF.
No trades could be made, order added to sell book.
70.00
Product bought in entirety, trades as follows:
trader1->trader2: 20.00xABC for $3.00.
No trades could be made, order added to buy book.
No such order in buy book.
Order successfully cancelled.
Order successfully cancelled.
No such order in sell book.
trader1
trader2
trader1->trader2: 20.00xABC for $3.00.
trader1->trader2: 20.00xABC for $3.00.
trader1->trader2: 20.00xABC for $3.00.
trader1->trader2: 20.00xABC for $3.00.
No trades have been completed with given product.
The sell book is empty.
The buy book is empty.
Have a nice day.